format-parquet/src/test/java/io/cdap/plugin/format/parquet/input/ParquetInputFormatProviderTest.java

```diff
@@ -74,4 +74,79 @@ public void testSchemaDetection() throws IOException {
+    GenericRecord record = new GenericRecordBuilder(avroSchema)
+      .set("id", 0)
+      .set("name", "alice")
+      .set("score", 1.0d)
+      .build();
+
+    ParquetWriter<GenericRecord> parquetWriter =
+      AvroParquetWriter.<GenericRecord>builder(HadoopOutputFile.fromPath(parquetPath, hConf))
+        .withSchema(avroSchema)
+        .build();
+    parquetWriter.write(record);
+    parquetWriter.close();
+
+    ParquetInputFormatProvider.Conf conf = new ParquetInputFormatProvider.Conf("pathField");
+    ParquetInputFormatProvider formatProvider = new ParquetInputFormatProvider(conf);
+    FormatContext formatContext = new FormatContext(new MockFailureCollector(), null);
+    SchemaDetector schemaDetector = new SchemaDetector(formatProvider);
+    Schema schema = schemaDetector.detectSchema(parquetFile.getAbsolutePath(), formatContext, Collections.emptyMap());
+    Assert.assertEquals(expectedSchema, schema);
+
+    //No Validation Failures Check
+    Assert.assertEquals(formatContext.getFailureCollector().getValidationFailures().size(), 0);
+
+  }
+
+
+  @Test
+  public void testPathFieldConflict() throws IOException {
+    Configuration hConf = new Configuration();
+    File parquetFile = new File(TMP_FOLDER.newFolder(), "test.parquet");
+    Path parquetPath = new Path(parquetFile.toURI());
+    Schema expectedSchema = Schema.recordOf("x",
+                                            Schema.Field.of("id", Schema.of(Schema.Type.INT)),
+                                            Schema.Field.of("name", Schema.of(Schema.Type.STRING)),
+                                            Schema.Field.of("score", Schema.of(Schema.Type.DOUBLE)),
+                                            Schema.Field.of("pathField", Schema.of(Schema.Type.STRING)));
+    org.apache.avro.Schema avroSchema = new org.apache.avro.Schema.Parser().parse(expectedSchema.toString());
+    GenericRecord record = new GenericRecordBuilder(avroSchema)
+      .set("id", 0)
+      .set("name", "alice")
+      .set("score", 1.0d)
+      .set("pathField", "value")
+      .build();
+    ParquetWriter<GenericRecord> parquetWriter =
+      AvroParquetWriter.<GenericRecord>builder(HadoopOutputFile.fromPath(parquetPath, hConf))
+        .withSchema(avroSchema)
+        .build();
+    parquetWriter.write(record);
+    parquetWriter.close();
+
+    ParquetInputFormatProvider.Conf conf = new ParquetInputFormatProvider.Conf("pathField");
+    ParquetInputFormatProvider formatProvider = new ParquetInputFormatProvider(conf);
+    FormatContext formatContext = new FormatContext(new MockFailureCollector(), null);
+    SchemaDetector schemaDetector = new SchemaDetector(formatProvider);
+    schemaDetector.detectSchema(parquetFile.getAbsolutePath(), formatContext, Collections.emptyMap());
+    ValidationFailure validationFailure = formatContext.getFailureCollector().getValidationFailures().get(0);
+    Assert.assertNotNull(validationFailure.withConfigProperty("pathField"));
```

Comment:
check that the value equals the expected value, not that it's not null

Reply:
What value to be checked?

Reply:
Changed to Checking value